### ISSUE TITLE
Stable?

### ISSUE THREAD

**Issue by the-maldridge** [protosam/go-libnss-etcd#1]
This looks pretty slick, I wish I'd had it available a year ago!  Right now [NetAuth](https://netauth.org) uses libnss_cache to do offline lookups.  Unfortunately libnss_cache has a number of glibc-isms, and makes it fairly difficult to use on musl systems.  Your pure Go implementation looks like it could be just the ticket to build both a portable version of libnss_cache, and to make an online version for NetAuth to back into.

Of course all of this depends on this library being stable, which would usually be signified by a tagged release.  In the absence of this, I ask here if this library is stable for production use.

**Comment by protosam** [comment]
This was a test case when I wrote it.

It has been working in my lab environment for the past 6 months, but that environment doesn't change often user-wise. My own sample size is really small.

I need more people to really do testing with it and collaborate on this, before I can really call it stable. I can do patching for this in my free time as I'm still using it.

If you needed some form of vendor support from me, I could work out some arrangement for that too.

**Comment by the-maldridge** [comment]
I don't think I'm looking for vendor support, mainly just that this library is reasonably stable and you don't intend to make any significant interface changes.  I was surprised to find commented out sections of code in the Go, so I wanted to check before I ported part of an application over to it.

I'm happy to collaborate, I'll send you patches as I go.

**Comment by protosam** [comment]
Yeah, no major interface changes will be made. All methods will retain the names they have unless for some reason libnss itself changes. 

Same deal with go-libnss in that regard. 

If you see room for any improvements, definitely open an issue for discussion or throw in a pull request. My only real commit criteria requires that:
- It adds value
- It doesn't break existing code (without good reason)

These are still repos that I use in my test deployments. :)
